### PR TITLE
Add validation and tests

### PR DESCRIPTION
**Pull request by Surafels**
Hello 👋 

In this branch, the following are implemented:

**The following validations are added:**

   - [ ] For the User model:

- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

   - [ ] For the Post model:

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

- [ ] Add unit specs for all models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    expect(user).to_not be_valid
+    expect(user.errors[:posts_counter]).to include('is not a number')
+  end
+
+  it 'is invalid when posts counter is less than zero' do
+    user = User.new(name: 'John Isac', posts_counter: -1)
+    expect(user.valid?).to be false
+    expect(user.errors[:posts_counter]).to include('must be greater than or equal to 0')
+  end
+end
```
Comment (ridwanediallo) [inline spec/models/user_spec.rb]: - [ ] Great job adding the validations and testing them 👏. However, as stated in the requirement, all your models' methods should be tested too. So please, I would suggest you test the `recent_posts` method.

### Requirement :point_down: 
Screenshot from 2023-12-15 16-53-41

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    expect(post).to_not be_valid
+    expect(post.errors[:likes_counter]).to include('is not a number')
+  end
+
+  it 'is invalid when likes counter is less than zero' do
+    post = Post.new(title: 'John Isaac', likes_counter: -1)
+    expect(post).to_not be_valid
+    expect(post.errors[:likes_counter]).to include('must be greater than or equal to 0')
+  end
+end
```
Comment (ridwanediallo) [inline spec/models/post_spec.rb]: - [ ] Great job adding the validations and testing them 👏. However, as stated in the requirement, all your models' methods should be tested too. So please, I would suggest you test the `update_user_posts_counter` and `recent_comments` methods.

### Requirement :point_down: 
Screenshot from 2023-12-15 16-53-41

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  it 'should be valid when likes_counter is zero' do
+    post.likes_counter = 0
+    expect(post).to be_valid
+  end
+
+  it 'should be valid when likes_counter is positive' do
+    post.likes_counter = 2
+    expect(post).to be_valid
+  end
+end
```
Comment (ridwanediallo) [inline spec/models/like_spec.rb]: - [ ] Great job adding the validations and testing them 👏. However, as stated in the requirement, all your models' methods should be tested too. So please, I would suggest you test the `update_likes_counter` method.

### Requirement :point_down: 
Screenshot from 2023-12-15 16-53-41

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      post.comments_counter = 0
+      expect(post).to be_valid
+    end
+
+    it 'should be valid when comments_counter is positive' do
+      post.comments_counter = 1
+      expect(post).to be_valid
+    end
+  end
+end
```
Comment (ridwanediallo) [inline spec/models/comment_spec.rb]: - [ ] Great job adding the validations and testing them 👏. However, as stated in the requirement, all your models' methods should be tested too. So please, I would suggest you test the `update_post_comments_counter` method.

### Requirement :point_down: 
Screenshot from 2023-12-15 16-53-41

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @Surafels , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- You have successfully added the validations and tested them. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @Surafels, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implemented the test for the models methods ✅ 
image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**